lsp/server/src/diagnostic/DiagnosticProducer.ts

```diff
@@ -19,4 +26,9 @@ export interface DiagnosticProducer<T> {
         textDocument: TextDocument,
         astNode: T
```

Comment:
Like we discussed in the other [PR](https://github.com/salesforce/salesforcedx-vscode-mobile/pull/127#discussion_r1805518966) you should make `astNode` a more generic name and probably a optional parameter.

Reply:
sure, I will name it 'data'

Reply:
done